<launch>
<!--Run the data form the rosbag and display it on Rviz -->
    <arg name="rosbag_folder" default="/home/morgan/Documents/Code_these/rosbags/"/>

    <!-- Select the rosbag_file -->
<!--    <arg name="rosbag_file" default="2rovs_tests_sur_penfeld_octobre_2023/clyde_without_cam__2023-10-03-15-42-34/2023-10-03-15-42-35.bag"/>-->
<!--    <arg name="rosbag_file" default="2rovs_tests_sur_penfeld_octobre_2023/face_a_face_echec_bizarre_2023-10-03-15-02-52/2023-10-03-15-02-53.bag"/>-->
<!--    <arg name="rosbag_file" default="2rovs_tests_sur_penfeld_octobre_2023/formation_triangle_test1_sans_orientation__2023-10-03-14-28-18/2023-10-03-14-28-19.bag"/>-->
<!--    <arg name="rosbag_file" default="2rovs_tests_sur_penfeld_octobre_2023/sac_de_noeud/2023-10-03-14-33-17.bag"/>-->
<!--    <arg name="rosbag_file" default="2rovs_tests_piscine_24oct2023/gain_faible/2023-10-24-16-49-10.bag"/>-->
<!--    <arg name="rosbag_file" default="2rovs_tests_piscine_24oct2023/gain_fort/2023-10-24-16-36-30.bag"/>-->
<!--    <arg name="rosbag_file" default="2rovs_tests_piscine_24oct2023/video_pour_lucia_piscine/2023-10-24-16-45-03.bag"/>-->

<!--    <arg name="rosbag_file" default="drone_and_cap_2023/blinky_without_cam_2023-11-06-18-31-22/2023-11-06-18-31-23.bag"/>-->
<!--    <arg name="rosbag_file" default="drone_and_cap_2023/inky_without_cam_2023-11-06-17-11-42/2023-11-06-17-11-42.bag"/>-->
<!--    <arg name="rosbag_file" default="drone_and_cap_2023/inky_without_cam_2023-11-06-17-21-55/2023-11-06-17-21-56.bag"/>-->

    <!-- Virtual Structure -->
    <arg name="rosbag_file" default="drone_and_cap_2023/inky_without_cam_2023-11-06-17-48-13/2023-11-06-17-48-14.bag"/>

<!--    <arg name="rosbag_file" default="drone_and_cap_2023/blinky_2023-11-08-15-14-22/2023-11-08-15-14-23.bag"/>-->
<!--    <arg name="rosbag_file" default="drone_and_cap_2023/blinky_2023-11-08-15-14-58/2023-11-08-15-14-59.bag"/>-->
<!--    <arg name="rosbag_file" default="drone_and_cap_2023/blinky_without_cam_2023-11-08-15-32-15/2023-11-08-15-32-16.bag"/>-->

    <!-- Leader Follower -->
<!--    <arg name="rosbag_file" default="drone_and_cap_2023/inky_without_cam_2023-11-08-15-25-03/2023-11-08-15-25-03.bag"/>-->

    <!--    <arg name="rosbag_file" default="drone_and_cap_2023/inky_without_cam_2023-11-08-15-28-37/2023-11-08-15-28-38.bag"/>-->
    <!--    <arg name="rosbag_file" default="drone_and_cap_2023/inky_without_cam_2023-11-08-16-05-59/2023-11-08-16-06-00.bag"/>-->


    <arg name="rvizconfig" default="$(find rov_rviz)/rviz/2rovs.rviz" />
    <arg name ="rovA_name" default="inky"/>
    <arg name ="rovB_name" default="blinky"/>
    <param name="rovA_name" value="$(arg rovA_name)"/>
    <param name="rovB_name" value="$(arg rovB_name)"/>

    <!-- set the value of the heading of the origin (fixed USBL) -->
    <!-- -288 for 2rovs_tests_sur_penfeld_octobre_2023 -->
    <!-- -10 for 2rovs_tests_piscine_24oct2023 and drone_and_cap_2023 -->
    <param name="cap_usbl_deg" value="-10"/>

    <node pkg="rosbag" type="play" name="rosbag" args="$(arg rosbag_folder)$(arg rosbag_file) --topics
    /usbl /$(arg rovA_name)/mavros/global_position/local /$(arg rovB_name)/mavros/global_position/local
    /$(arg rovA_name)/coord_objectif /$(arg rovB_name)/coord_objectif
    /$(arg rovA_name)/mavros/rc/override /$(arg rovB_name)/mavros/rc/override"/>
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />
    <node name="rov_rviz" pkg="rov_rviz" type="rov_rviz_node" required="true" output="screen" />
</launch>
